lib/puppet/info_service/task_information_service.rb

```diff
@@ -0,0 +1,25 @@
+class Puppet::InfoService::TaskInformationService
+
+  def self.tasks_per_environment(environment_name)
+    # get the actual environment object, raise error if the named env doesn't exist
+    env = Puppet.lookup(:environments).get!(environment_name)
+    env.modules.map do |mod|
+      mod.tasks.map do |task|
+        {:module => {:name => task.module.name}, :name => task.name}
```

Comment:
Room to add more information to the module to e.g. disambiguate between different versions in different environments.